flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecRank.java

```diff
@@ -207,18 +209,35 @@ public StreamExecRank(
 
         AbstractTopNFunction processFunction;
         if (rankStrategy instanceof RankProcessStrategy.AppendFastStrategy) {
-            processFunction =
-                    new AppendOnlyTopNFunction(
-                            minIdleStateRetentionTime,
-                            maxIdleStateRetentionTime,
-                            inputRowTypeInfo,
-                            sortKeyComparator,
-                            sortKeySelector,
-                            rankType,
-                            rankRange,
-                            generateUpdateBefore,
-                            outputRankNumber,
-                            cacheSize);
+            if (sortFields.length == 1
+                    && inputType.getChildren().get(sortFields[0]).getTypeRoot()
+                            == LogicalTypeRoot.TIMESTAMP_WITHOUT_TIME_ZONE
```

Comment:
Just `TIMESTAMP_WITHOUT_TIME_ZONE`? I don't see any `proc-time`.

Reply:
sorry, forgot to push the latest commit.